src/tests/test_vesync_bulbs.py

```diff
@@ -78,10 +95,233 @@ def test_invalid_brightness(self, caplog, api_mock):
+        self.vesync_obj.get_devices()
+        bulbs = self.vesync_obj.bulbs
+        assert len(bulbs) == 1
+        bulb = bulbs[0]
+        assert isinstance(bulb, VeSyncBulbESL100MC)
+        assert bulb.device_name == 'ESL100MC NAME'
+        assert bulb.device_type == 'ESL100MC'
+        assert bulb.cid == 'CID-ESL100MC'
+        assert bulb.uuid == 'UUID-ESL100MC'
+
+    def test_esl100mc_details(self, api_mock):
+        """Test WS get_details()."""
+        self.mock_api.return_value = DEV_LIST_MC
+        bulb = VeSyncBulbESL100MC(DEV_LIST_DETAIL_MC, self.vesync_obj)
+        bulb.get_details()
+        assert self.mock_api.r
+        assert bulb.device_status == 'on'
+        assert bulb.connection_status == 'online'
+
+    def test_esl100mc_onoff(self, caplog, api_mock):
+        """Test power toggle for ESL100 bulb."""
+        self.mock_api.return_value = ({'code': 0}, 200)
+        bulb = VeSyncBulbESL100MC(DEV_LIST_DETAIL_MC, self.vesync_obj)
+        assert bulb.turn_off()
+        assert bulb.device_status == 'off'
+        assert bulb.turn_on()
+        assert bulb.device_status == 'on'
+
+    def test_brightness(self, api_mock):
+        self.mock_api.return_value = ({'code': 0}, 200)
+        bulb = VeSyncBulbESL100MC(DEV_LIST_DETAIL_MC, self.vesync_obj)
+        assert bulb.set_brightness(50)
+        assert bulb.brightness == 50
+
+    def test_invalid_brightness(self, caplog, api_mock):
+        self.mock_api.return_value = ({'code': 0}, 200)
+        bulb = VeSyncBulbESL100MC(DEV_LIST_DETAIL_MC, self.vesync_obj)
+        assert bulb.set_brightness(5000)
+        assert bulb.brightness == 100
+
+    def test_color(self, api_mock):
+        self.mock_api.return_value = ({'code': 0}, 200)
+        bulb = VeSyncBulbESL100MC(DEV_LIST_DETAIL_MC, self.vesync_obj)
+        assert bulb.set_rgb(50, 100, 150)
+        assert bulb.color_rgb == namedtuple('rgb', 'red green blue')(50, 100, 150)
+
+    def test_features(self, api_mock):
+        bulb = VeSyncBulbESL100MC(DEV_LIST_DETAIL_MC, self.vesync_obj)
+        assert bulb.dimmable_feature
+        assert not bulb.color_temp_feature
+        assert bulb.rgb_shift_feature
+
+class TestVeSyncBulbValceno:
+    """Tests for VeSync Valceno bulb."""
+
+    @pytest.fixture()
+    def api_mock(self, caplog):
+        """Initilize the mock VeSync class and patch call_api."""
+        self.mock_api_call = patch('pyvesync.helpers.Helpers.call_api')
+        self.mock_api = self.mock_api_call.start()
+        self.mock_api.create_autospect()
+        self.mock_api.return_value.ok = True
+        self.vesync_obj = VeSync('sam@mail.com', 'pass')
+        self.vesync_obj.enabled = True
+        self.vesync_obj.login = True
+        self.vesync_obj.token = 'sample_tk'
+        self.vesync_obj.account_id = 'sample_actid'
+        caplog.set_level(logging.DEBUG)
+        yield
+        self.mock_api_call.stop()
+
+    def test_esl100vc_conf(self, api_mock):
```

Comment:
Why `esl100` naming here? This name doesn't have anything to do with this bulb in particular. It could start some confusion. Should it be this model code instead ? `XYD0001`